pkg/operation/botanist/kubeapiserver.go

```diff
@@ -334,12 +334,31 @@ func (b *Botanist) computeKubeAPIServerServerCertificateConfig() kubeapiserver.S
 	}
 }
 
+const annotationKeyNewEncryptionKeyPopulated = "credentials.gardener.cloud/new-encryption-key-populated"
+
 func (b *Botanist) computeKubeAPIServerETCDEncryptionConfig(ctx context.Context) (kubeapiserver.ETCDEncryptionConfig, error) {
 	config := kubeapiserver.ETCDEncryptionConfig{
 		RotationPhase:         gardencorev1beta1helper.GetShootETCDEncryptionKeyRotationPhase(b.Shoot.GetInfo().Status.Credentials),
 		EncryptWithCurrentKey: true,
 	}
 
+	if gardencorev1beta1helper.GetShootETCDEncryptionKeyRotationPhase(b.Shoot.GetInfo().Status.Credentials) == gardencorev1beta1.RotationPreparing {
+		deployment := &metav1.PartialObjectMetadata{}
+		deployment.SetGroupVersionKind(appsv1.SchemeGroupVersion.WithKind("Deployment"))
+		if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, v1beta1constants.DeploymentNameKubeAPIServer), deployment); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return kubeapiserver.ETCDEncryptionConfig{}, err
+			}
+		}
+
+		// If the new encryption key was not yet populated to all replicas then we should not still use the old key for
```

Comment:
```suggestion
		// If the new encryption key was not yet populated to all replicas then we should still use the old key for
```

To my understanding if the new encryption key is not yet added to the EncryptionConfiguration, then the old key should be still used for encrypting the data. Right?

Reply:
Yes, right, thanks!